login/src/main/java/com/fightpandemics/login/ui/profile/CompleteProfileFragment.kt

```diff
@@ -6,102 +6,165 @@ import android.os.Bundle
-            val otherHelp: Boolean =
-                rootView.findViewById<MaterialCheckBox>(R.id.other_help_checkbox).isChecked
-
-            val objectives = Objectives(donation, information, volunteerHrsOffering)
-            val needs = Needs(volunteerHrsRequest, otherHelp)
-            val hide = Hide(false)
-            val location = Location("mock", "mock", listOf(0.0, 0.0), "mock", "mock")
-
-            val completeProfileRequest = CompleteProfileRequest(firstName, hide, lastName, location, needs, objectives)
-            onCompleteProfile(completeProfileRequest)
+        savedInstanceState: Bundle?,
+    ): View {
+        _fragmentCompleteProfileBinding = FragmentCompleteProfileBinding.inflate(inflater)
+
+        fragmentCompleteProfileBinding.completeProfileToolbar.root.setOnClickListener {
+            findNavController().navigateUp()
+        }
+
+        fragmentCompleteProfileBinding.clBtnComplete.setOnClickListener {
+            validateAndCompleteProfile()
         }
 
-        return rootView
+        setupLocationAutocomplete()
+        setupShareLocation()
+
+        return fragmentCompleteProfileBinding.root
     }
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         arguments?.getString(USER_PROFILE)
     }
 
-    @ExperimentalCoroutinesApi
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _fragmentCompleteProfileBinding = null
+    }
+
+    private fun validateAndCompleteProfile() {
+        val firstName: String = fragmentCompleteProfileBinding.etFirstName.text.toString()
+        val lastName: String = fragmentCompleteProfileBinding.etLastName.text.toString()
+        val donation: Boolean = fragmentCompleteProfileBinding.root.donation_checkbox.isChecked
+        val information: Boolean =
+            fragmentCompleteProfileBinding.root.information_checkbox.isChecked
+        val volunteerHrsOffering: Boolean =
+            fragmentCompleteProfileBinding.root.hours_offer_checkbox.isChecked
+        val volunteerHrsRequest: Boolean =
+            fragmentCompleteProfileBinding.root.hours_request_checkbox.isChecked
+        val otherHelp: Boolean = fragmentCompleteProfileBinding.root.other_help_checkbox.isChecked
+        val objectives = Objectives(donation, information, volunteerHrsOffering)
+        val needs = Needs(volunteerHrsRequest, otherHelp)
+        val hide = Hide(false)
+
+        if (isFirstNameValid(firstName) and isLastNameValid(lastName) and isLocationValid()) {
+            val location = loginViewModel.completeProfileLocation
+            val completeProfileRequest =
+                CompleteProfileRequest(firstName, hide, lastName, location, needs, objectives)
+            Timber.i("my complete profile request is $completeProfileRequest")
+            onCompleteProfile(completeProfileRequest)
+        }
+    }
+
+    private fun isLocationValid(): Boolean {
```

Comment:
Perhaps rename this method since it not only checks if location is valid but also has some side effects. At least if I understand correctly what the error assignment does.